src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseFileRepository.cs

```diff
@@ -99,6 +102,11 @@ public async Task<ReleaseFile> Get(Guid releaseId, Guid fileId)
                     && releaseFile.FileId == fileId);
         }
 
+        public async Task<Either<ActionResult, ReleaseFile>> FindOrNotFound(Guid releaseId, Guid fileId)
```

Comment:
I think we already have this Find / FindOrNotFound pattern for free within AbstractRepository<T> - you could look to extend it and see if it gives you what you need.  It might give you more than you need too!

Reply:
I thought about this, but unfortunately it's only in the `Data.Services` project and is currently specific for the `StatisticsDbContext`. Thought it best to avoid re-purposing it here as it's used in a number of places.

It would be good if we could create a common base repository class for stuff like this that we could have in `Common`. I've created a ticket for it anyway: https://dfedigital.atlassian.net/browse/EES-2529

Reply:
Cool ta